### ISSUE TITLE
Unexpected behaviors for --size-for-MAG and --min-completion-for-MAG in anvi-rename-bins

### ISSUE THREAD

**Issue by jmiyoshi** [merenlab/anvio#675]
Hi,

I'm using Anvi'o ver. 3.  I experienced unexpected behaviors in the program, anvi-rename-bins.
I tried to define metagenomic bins with >70% completion and >2 Mbp in length as MAGs and rename the MAGs. However, some bins that satisfied this criterion did not get renamed as MAGs and other bins that did not satisfy this criterion were renamed as MAGs.

FYI, what I ran was;

$ anvi-rename-bins -c CONTIGS.db -p PROFILE.db --collection-to-read CONCOCT_c25 --collection-to-write FINAL --use-SCG-averages --call-MAGs --size-for-MAG 2 --min-completion-for-MAG 70 --max-redundancy-for-MAG 10 --prefix O_PUP --report-file FINAL.txt

Thank you.

Jun

**Comment by meren** [comment]
This sounds very important and we should look into this before the new release.

**Comment by tdelmont** [comment]
One part of the problem might be due to the bins displaying high levels of
redundancy:

Here is the rule:
---
*  --max-redundancy-for-MAG [0-100]*
*                        If --call-MAGs flag is used, call any bin a 'MAG'
if*
*                        their redundancy estimate is below this (the
default*
*                        is 10) and the completion estimate is above --min-*
*                        completion-for-MAG.*
---

Jun, could you please make sure the unexpected behaviors in not linked to
bins with redundancy > 10%.

Regarding the bins unexpectedly renamed as MAGs, could you please provide
an example that includes length, completion and redundancy?

Thank you,

Tom

**Comment by jmiyoshi** [comment]
Hi Tom,
I doublechecked my result with anvi-interactive collection mode and confirmed the unexpected behaviors (some bins do not satisfy the criterion get renamed as MAGs) occur in bins with redundancy <10%. For example, one bin named "MAG" has (1) length: 2910588, (2) completion: 37.4%, and (3) redundancy: 5.03%, on the other hand, another "MAG" bin has (1) length: 2278657, (2) completion: 98.5%, and (3) redundancy: 1.43%. I also found a bin with (1) length: 2645767, (2) completion: 89.9%, and (3) redundancy: 10.7% is renamed as "Bin" correctly.
It seems Anvi'o does not take % of completion into account, but I'm not sure.
Thank you so much!
Jun
